src/v/cloud_storage/remote_partition.cc

```diff
@@ -242,12 +243,25 @@ class partition_record_batch_reader_impl final
         if (config.abort_source) {
             vlog(_ctxlog.debug, "abort_source is set");
             _partition_reader_as = config.abort_source;
-            auto sub = config.abort_source->get().subscribe([this]() noexcept {
-                vlog(_ctxlog.debug, "abort requested via config.abort_source");
-                if (_seg_reader) {
-                    _partition->evict_segment_reader(std::move(_seg_reader));
-                }
-            });
+            auto sub = config.abort_source->get().subscribe(
+              [this](const std::optional<std::exception_ptr>& eptr) noexcept {
+                  auto reason = eptr.has_value()
+                                  ? net::is_disconnect_exception(*eptr)
+                                  : "shutdown";
+                  if (reason) {
+                      vlog(
+                        _ctxlog.debug,
+                        "abort requested via config.abort_source: {}",
+                        reason);
+                  } else {
+                      vlog(
+                        _ctxlog.debug,
+                        "abort requested via config.abort_source");
+                  }
+                  if (_seg_reader) {
+                      _partition->evict_segment_reader(std::move(_seg_reader));
+                  }
```

Comment:
i think we are missing the case where there is an exception but it's not a disconnect_exception.

if we want to increase coverage, 

```cpp
if(!eptr.has_value()){
    vlog(_ctxlog.debug, "abort requested via config.abort_source");
} else if(auto maybe_reason= net::is_disconnect_exception(*eptr);
    maybe_reason.has_value()){
    vlog(_ctxlog.debug, "abort requested via config.abort_source: {}", *maybe_reason);
} else {
    vlog(_ctxlog.debug, "abort requested via config.abort_source: {}", *eptr);
}    
```

Reply:
@Lazin in your e-mail response to @andijcr you wrote:

>In this case the exception is just rethrown. Feel free to send another PR
or push commits to this one of it's needed for release.

but it's not clear what you mean by `In this case the exception is just rethrown` it doesn't look like anything is rethrown now (or before).

afaict the change has exactly the same error handling semantics as before, but there is some better logging, and I think Andrea is only mentioning that there is a third scenario for the log which could provide additional context.

I'm going to go ahead and merge this since it looks fine to me, but please update the logging here if we think that the extra condition should be logged.